### ISSUE TITLE
Python Scalars Support

### ISSUE THREAD

**Issue by nmichlo** [jonasrauber/eagerpy#32]
**Feature Request**

Would is be possible to support python scalars?

eg.
```python3
num_float = ep.astensor(1.0)
num_int = ep.astensor(42)
```

**Why?**

This would be helpful for implementing generic functions, for example:

```python3
def get_kernel_size(sigma=1.0, trunc=4.0):
    sigma, trunc = ep.astensors(sigma, trunc)
    radius = (sigma * trunc + 0.5).astype(int)
    return (2 * radius + 1).raw
```

This could be called with default values as well as tensors.

```python3
t_sigma = torch.abs(torch.randn(10))
n_truch = np.abs(np.random.randn(10))

get_kernel_size(sigma=1.0, trunc=4.0)
get_kernel_size(sigma=t_sigma, trunc=4.0)
get_kernel_size(sigma=1.0, trunc=n_truch)
```

**Comment by jonasrauber** [comment]
Nice example. I see why this can be useful. It might be possible to add a simple wrapper class for python scalars that automatically gets unwrapped when it interacts with any tensor.

**Comment by nmichlo** [comment]
That sounds ideal.

I am not sure about the internals of the library, but I imagine this could have implications for the various helper methods and functions? I am not sure how these would or could be handled?

**Comment by jonasrauber** [comment]
Yes, I think there are quite a few cases to consider and some corner cases might be challenging, but it should be doable.

For now, this workaround should be fine (if you assume that always at least one of the arguments is a tensor):
```python
def get_kernel_size(sigma=1.0, trunc=4.0):
    if isinstance(sigma, int) or isinstance(sigma, float):
        trunc = ep.astensor(trunc)
        sigma = ep.zeros(trunc, ()) + sigma
    elif isinstance(trunc, int) or isinstance(trunc, float):
        sigma = ep.astensor(sigma)
        trunc = ep.zeros(sigma, ()) + trunc
    else:
        sigma, trunc = ep.astensors(sigma, trunc)
    radius = (sigma * trunc + 0.5).astype(int)
    return (2 * radius + 1).raw
```
(I didn't test this)


You can introduce helper functions to simplify it further.

In fact, we could actually integrate all of the above into `ep.astensors`. It works as long as one of the arguments as actually a tensor (i.e., not all of them can be ints and floats).

**Comment by peekxc** [comment]
More generally, you might want to consider doing instance checks against the appropriate [number type](https://docs.python.org/3/library/numbers.html)
